tilings/tiling.py

```diff
@@ -933,28 +936,28 @@ def gridded_perms(self,
         The gridded permutations are up to length of the longest minimum
         gridded permutations that is griddable on the tiling.
         """
-        return GriddedPermsOnTiling(self, maxlen=maxlen)
+        yield from GriddedPermsOnTiling(self, maxlen=maxlen)
 
-    def merge(self, remove_empty=True):
+    def merge(self) -> 'Tiling':
         """Return an equivalent tiling with a single requirement list."""
         if len(self.requirements) <= 1:
             return self
-        reqs = sorted(self.requirements, key=len)
-        req1 = reqs[0]
-        req2 = reqs[1]
-        reqs = reqs[2:]
-        new_req = []
-        for gp1 in req1:
-            for gp2 in req2:
-                # TODO: Do this step independent of tilings.
-                temp_tiling = Tiling(self.obstructions, [[gp1], [gp2]],
-                                     remove_empty=False)
-                new_req.extend(Requirement(gp.patt, gp.pos)
-                               for gp in temp_tiling.gridded_perms(
-                    maxlen=len(gp1) + len(gp2)))
-        merged_tiling = Tiling(self.obstructions, reqs + [new_req],
-                               remove_empty=remove_empty)
-        return merged_tiling
+        mgps = MinimalGriddedPerms(self)
+        requirements = tuple(Requirement(gp.patt, gp.pos) for gp in
+                             mgps.minimal_gridded_perms())
+        return self.__class__(self.obstructions, (requirements,))
+
+    def minimal_gridded_perms(self, yield_non_minimal=False):
```

Comment:
Having this flag is kind of weird. I would remove it and change `is_empty` to
```python
MGP = MinimalGriddedPerms(self)
return any(True for _ in MGP.minimal_gridded_perms(yield_non_minimal))
```

Reply:
Agree, fixed.